core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -3364,6 +3364,20 @@ && isHourKeyword(columnExpr.token)
                         tempVaf.getQuick(i).pushValueTypes(arrayColumnTypes);
                     }
 
+                    if (tempVaf.size() == 0) {// select distinct case 
+                        ArrayColumnTypes columnTypes = new ArrayColumnTypes();
```

Comment:
creating new list seems unnecessary, if i make this change:
![image](https://github.com/questdb/questdb/assets/7276403/b7d3eb1a-acdf-45de-8595-b95f15a1a8d2)

tests pass also